src/Azure/Azure.Quantum.Client/SubmitterFactory.cs

```diff
@@ -0,0 +1,123 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+#nullable enable
+
+namespace Microsoft.Azure.Quantum
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Collections.Immutable;
+    using System.Linq;
+    using System.Reflection;
+    using System.Text.RegularExpressions;
+    using Microsoft.Quantum.Runtime.Submitters;
+
+    /// <summary>
+    /// Creates Azure Quantum submitters for specific targets.
+    /// </summary>
+    public static class SubmitterFactory
+    {
+        /// <summary>
+        /// Information about each supported QIR submitter.
+        /// </summary>
+        private static readonly ImmutableList<SubmitterInfo> QirSubmitters = ImmutableList.Create(
+            new SubmitterInfo(
+                new Regex(@"^microsoft\.simulator$"),
+                "Microsoft.Quantum.Providers.Targets.MicrosoftSimulatorSubmitter, Microsoft.Quantum.Providers.Core",
```

Comment:
Is the first part (before the `,`) the name of the type and the second part the assembly name? Is this how fully-qualified names are defined?

Reply:
Yes, it's an [assembly-qualified name](https://docs.microsoft.com/en-us/dotnet/api/system.type.assemblyqualifiedname?view=netcore-3.1).